### ISSUE TITLE
how to draw a vertical line from known x axis to kaplan meier curve and from the intersection point to y axis

### ISSUE THREAD

**Issue by Siahrood** [kassambara/survminer#320]
Hi , Is that possible to draw a line from known x-axis pont to Survival graph and from intersection point a line to y-axis in "survival curve based on kaplan meier estimate" ? 
I have found by "geom_segment" it is possible but I guess in that case y-axis points are known but in my case y-axis can be define from intersection of the first line with survival graph.

could you please help me in this regard ?
best regards
syahrood

**Comment by kassambara** [comment]
Yes. Read the survival probability off the fit at your known x with `summary(fit, times = x)$surv`, then draw the two dashed segments with `annotate("segment", ...)` — the y-coordinate comes from that lookup, so you don't need to know it in advance:

```r
library(survival); library(survminer); library(ggplot2)
fit <- survfit(Surv(time, status) ~ 1, data = lung)

xt <- 365                                  # known x
yt <- summary(fit, times = xt)$surv        # S(xt), read off the curve  -> 0.409

p <- ggsurvplot(fit, data = lung, conf.int = FALSE, legend = "none")
p$plot <- p$plot +
  annotate("segment", x = xt, y = 0,  xend = xt, yend = yt, linetype = "dashed", colour = "red") +
  annotate("segment", x = 0,  y = yt, xend = xt, yend = yt, linetype = "dashed", colour = "red") +
  annotate("point",   x = xt, y = yt, colour = "red", size = 2)
p
```

![vertical + horizontal line to the KM curve](https://i.imgur.com/pFUWgrf.png)

The vertical line rises from x = 365 to the curve, and the horizontal line runs from that intersection to the y-axis (S ≈ 0.41). For grouped fits, call `summary(fit, times = xt)$surv` per stratum. Closing as answered.
